### DISCUSSION TITLE
Show only user goroutines by default

### DISCUSSION THREAD

**Discussion by suzmue** [golang/vscode-go#1754] [Discussions]
Using a debugger to debug a Go program reveals a lot of the Go runtime implementation details to the user. Several goroutines that belong to the runtime appear in the call stack, cluttering the view and making a simple program seem very complicated.

For example when debugging a simple hello world program, 5 different goroutines appear in the callstack: 
Program stopped on a breakpoint in `main.go`:
<img width="1078" alt="Screen Shot 2021-09-03 at 2 28 41 PM" src="https://user-images.githubusercontent.com/6634754/132394406-d9a97033-9ca9-4792-8278-fe942850fd95.png">

Call stack:
<img width="701" alt="Screen Shot 2021-09-03 at 2 28 48 PM" src="https://user-images.githubusercontent.com/6634754/132394421-b5b91393-d11e-4706-bf1a-78fc183e64e2.png">

This proposal is to (by default) only show the user goroutines in the call stack. When using this new default, the call stack for hello world will only show a single goroutine, as would be expected from a simple program with no concurrency:

<img width="497" alt="Screen Shot 2021-09-07 at 12 36 41 PM" src="https://user-images.githubusercontent.com/6634754/132394424-e67dafc0-a7fa-4b74-9a15-2402eed77215.png">

This would be controlled by a setting to allow users that really want to see the system goroutines to include them using the launch configuration.

**Comment by hyangah** [comment]
I don't see any issue or objection, but positive votes. Let's move to the issue tracker and make it work. 

Created #1797
